docs/source/parameter_handling.rst

```diff
@@ -39,4 +39,51 @@ The parameters can then be accessed as
     [('param1', 7.0), ('param2', 4.0)]
     >>> p.exists("param1")
     True
+    
+    
+.. code-block:: python
+ 
+    # print all data inside Param
+    def printAll(x):
+      if (x.size()):
+        for i in x.keys():
+          print("Key: ", i, "Value: ", x[i])
+
+      else :
+          print("no data availabe")
+
+
+    new_p = Param()
+    if (p.empty() == False): #check p is not empty
```

Comment:
is this check needed?
is new_p a soft copy? or deep copy? e.g. will changing new_p also change p? ;)